Mage.Sets/src/mage/cards/g/GollumSchemingGuide.java

```diff
@@ -0,0 +1,139 @@
+            "remove Gollum from combat. Otherwise, you draw a card and Gollum " +
+            "can't be blocked this turn";
+    }
+
+    private GollumSchemingGuideEffect(final GollumSchemingGuideEffect effect){
+        super(effect);
+    }
+
+    @Override public GollumSchemingGuideEffect copy(){
+        return new GollumSchemingGuideEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        Cards cards = new CardsImpl(player.getLibrary().getTopCards(game, 2));
+        // look at the top two cards of your library,
+        player.lookAtCards(source, null, cards, game);
+        // put them back in any order
+        PutCards.TOP_ANY.moveCards(player, cards, source, game);
+
+
+        // then choose land or nonland.
+        boolean guessLand = player.chooseUse(Outcome.Neutral,
+            "What should the opponent guess?",
+            "", "land", "nonland", source, game);
+
+        // choose an opponent.
+        TargetOpponent targetOpponent = new TargetOpponent();
+        targetOpponent.setNotTarget(true);
+        if (!player.choose(Outcome.Neutral, targetOpponent, source, game)) {
+            return false;
+        }
+
+        Player opponent = game.getPlayer(targetOpponent.getFirstTarget());
+        if (opponent == null) {
+            return false;
+        }
+
+        String textGuess = guessLand ? "land" : "nonland";
+
+        FilterCard filter = guessLand
+            ? StaticFilters.FILTER_CARD_LAND
+            : StaticFilters.FILTER_CARD_NON_LAND;
+
+        // An opponent guesses whether the top card of your library is the chosen kind.
+        boolean guessOpp = opponent.chooseUse(Outcome.Neutral,
+            "Is the top card of " + player.getLogName() + "'s library a " + textGuess + " card?", source, game);
+
+        // End of the choices, last check on whether the player still exist.
+        Card card = player.getLibrary().getFromTop(game);
+        player.revealCards(source, new CardsImpl(card), game);
+
+        if (card != null && (filter.match(card, player.getId(), source, game) == guessOpp)) {
+            game.informPlayers(opponent.getLogName() + " guessed right.");
+
+            // remove Gollum from combat.
+            Permanent gollum = game.getPermanent(source.getSourceId());
+            if (gollum != null) {
+                gollum.removeFromCombat(game);
+            }
+        } else {
+            game.informPlayers(opponent.getLogName() + " guessed wrong.");
+
+            // you draw a card
+            player.drawCards(1, source, game);
+
+            // and Gollum can't be blocked this turn
+            Permanent gollum = game.getPermanent(source.getSourceId());
+            if (gollum != null) {
+                game.addEffect(new CantBeBlockedSourceEffect(Duration.EndOfTurn), source);
```

Comment:
This is not adding any hint text on Gollum for "{this} can't be blocked this turn", although the effect is active.
do we have a convenient way to add those types of continuous effect hint?

Reply:
Found #7046 that reported similar issue with "can't be blocked this turn" not having any feedback.

Reply:
I'll try to improve the 'can't be blocked this turn' feedbacks in a separate PR.